cloud/services/networking/loadbalancers.go

```diff
@@ -43,6 +43,41 @@ func (s *Service) GetLoadBalancer(id string) (*godo.LoadBalancer, error) {
 
 // CreateLoadBalancer creates a LB.
 func (s *Service) CreateLoadBalancer(spec *infrav1.DOLoadBalancer) (*godo.LoadBalancer, error) {
+	request := s.loadBalancerRequest(spec)
+	lb, _, err := s.scope.LoadBalancers.Create(s.ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	return lb, nil
+}
+
+// NeedsUpdate checks if the load balancer needs to be updated.
+// Example: for a migrated docluster, the LB name and tag needs to be updated to reflect the new capi cluster's uid.
+func (s *Service) NeedsUpdate(lb *godo.LoadBalancer) bool {
+	return lb.Tag != s.getLBTag()
+}
```

Comment:
The comment mentions the name and tag need to be updated, however we only check the tag here. Is this intentional?